misc/dist/linux/org.godotengine.Godot.xml

```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0"?>
+<mime-info xmlns="http://www.freedesktop.org/standards/shared-mime-info">
+  <mime-type type="application/x-godot-project">
+    <comment>Godot Engine project</comment>
+    <icon name="x-godot-project" />
```

Comment:
We don't provide icons with these names ready to install, so it won't work unless packagers reverse engineer what they're supposed to do with this.

Reply:
> Is `org.godotengine.godot.xml` a standardized name for MIME type definitions? That doesn't seem to match what I have installed locally:
> 
> ```
> $ ls /usr/share/mime/application/
> andrew-inset.xml                                                   vnd.openxmlformats-officedocument.presentationml.slide.xml       x-glade.xml                     x-pkcs7-certificates.xml
> annodex.xml                                                        vnd.openxmlformats-officedocument.presentationml.template.xml    x-gnucash.xml                   x-planperfect.xml
> atom+xml.xml                                                       vnd.openxmlformats-officedocument.spreadsheetml.sheet.xml        x-gnumeric.xml                  x-plasma.xml
> dicom.xml                                                          vnd.openxmlformats-officedocument.spreadsheetml.template.xml     x-gnuplot.xml                   x-pocket-word.xml
> ecmascript.xml                                                     vnd.openxmlformats-officedocument.wordprocessingml.document.xml  x-go-sgf.xml                    x-prjx.xml
> epub+zip.xml                                                       vnd.openxmlformats-officedocument.wordprocessingml.template.xml  x-gpx+xml.xml                   x-profile.xml
> geo+json.xml                                                       vnd.palm.xml                                                     x-graphite.xml                  x-pw.xml
> gml+xml.xml                                                        vnd.rar.xml                                                      x-grisbi.xml                    x-pyspread-bz-spreadsheet.xml
> gnunet-directory.xml                                               vnd.recordare.musicxml.xml                                       x-gtk-builder.xml               x-pyspread-spreadsheet.xml
> gpx+xml.xml                                                        vnd.recordare.musicxml+xml.xml                                   x-gtktalog.xml                  x-python-bytecode.xml
> gzip.xml                                                           vnd.rn-realmedia.xml                                             x-gzdvi.xml                     x-qemu-disk.xml
> illustrator.xml                                                    vnd.smaf.xml                                                     x-gz-font-linux-psf.xml         x-qpress.xml
> javascript.xml                                                     vnd.snap.xml                                                     x-gzpdf.xml                     x-qtiplot.xml
> jrd+json.xml                                                       vnd.sqlite3.xml                                                  x-gzpostscript.xml              x-quanta.xml
> json-patch+json.xml                                                vnd.squashfs.xml                                                 x-hdf.xml                       x-quattropro.xml
> json.xml                                                           vnd.stardivision.calc.xml                                        x-hfe-floppy-image.xml          x-quicktime-media-link.xml
> ld+json.xml                                                        vnd.stardivision.chart.xml                                       xhtml+xml.xml                   x-qw.xml
> mac-binhex40.xml                                                   vnd.stardivision.draw.xml                                        x-hwp.xml                       x-raw-disk-image.xml
> mathematica.xml                                                    vnd.stardivision.impress.xml                                     x-hwt.xml                       x-raw-disk-image-xz-compressed.xml
> mathml+xml.xml                                                     vnd.stardivision.mail.xml                                        x-ica.xml                       x-raw-floppy-disk-image.xml
> mbox.xml                                                           vnd.stardivision.math.xml                                        x-icq.xml                       x-remoting.xml
> metalink4+xml.xml                                                  vnd.stardivision.writer.xml                                      x-iff.xml                       x-renderdoc-capture.xml
> metalink+xml.xml                                                   vnd.sun.xml.base.xml                                             x-ipod-firmware.xml             x-resources.xml
> msword-template.xml                                                vnd.sun.xml.calc.template.xml                                    x-ips-patch.xml                 x-resourcesx.xml
> msword.xml                                                         vnd.sun.xml.calc.xml                                             x-ipynb+json.xml                x-riff.xml
> mxf.xml                                                            vnd.sun.xml.draw.template.xml                                    x-iso9660-appimage.xml          x-rpm.xml
> octet-stream.xml                                                   vnd.sun.xml.draw.xml                                             x-it87.xml                      x-ruby.xml
> oda.xml                                                            vnd.sun.xml.impress.template.xml                                 x-java-applet.xml               x-sami.xml
> ogg.xml                                                            vnd.sun.xml.impress.xml                                          x-java-archive.xml              x-saturn-rom.xml
> owl+xml.xml                                                        vnd.sun.xml.math.xml                                             x-java-jce-keystore.xml         x-sc.xml
> oxps.xml                                                           vnd.sun.xml.writer.global.xml                                    x-java-jnlp-file.xml            xsd.xml
> pdf.xml                                                            vnd.sun.xml.writer.template.xml                                  x-java-keystore.xml             x-sega-cd-rom.xml
> pgp-encrypted.xml                                                  vnd.sun.xml.writer.xml                                           x-java-pack200.xml              x-sega-pico-rom.xml
> pgp-keys.xml                                                       vnd.symbian.install.xml                                          x-java.xml                      x-sf2.xml
> pgp-signature.xml                                                  vnd.tcpdump.pcap.xml                                             x-jbuilder-project.xml          x-sf3.xml
> pkcs10.xml                                                         vnd.visio.xml                                                    x-karbon.xml                    x-sfz.xml
> pkcs12.xml                                                         vnd.wordperfect.xml                                              x-kcachegrind.xml               x-sg1000-rom.xml
> pkcs7-mime.xml                                                     vnd.youtube.yt.xml                                               x-kchart.xml                    x-shared-library-la.xml
> pkcs7-signature.xml                                                winhlp.xml                                                       x-kcsrc.xml                     x-sharedlib.xml
> pkcs8-encrypted.xml                                                x-7z-compressed.xml                                              x-kexi-connectiondata.xml       x-shar.xml
> pkcs8.xml                                                          x-abiword.xml                                                    x-kexiproject-shortcut.xml      x-shellscript.xml
> pkix-cert.xml                                                      x-ace.xml                                                        x-kexiproject-sqlite2.xml       x-shorten.xml
> pkix-crl.xml                                                       x-alz.xml                                                        x-kexiproject-sqlite3.xml       x-siag.xml
> pkix-pkipath.xml                                                   x-amiga-disk-format.xml                                          x-kformula.xml                  x-sln.xml
> postscript.xml                                                     x-amipro.xml                                                     x-kgeo.xml                      x-slp.xml
> prs.plucker.xml                                                    x-aportisdoc.xml                                                 x-kgetlist.xml                  xslt+xml.xml
> raml+yaml.xml                                                      x-apple-diskimage.xml                                            x-khtml-adaptor.xml             x-smb-server.xml
> ram.xml                                                            x-apple-systemprofiler+xml.xml                                   x-kig.xml                       x-smb-workgroup.xml
> rdf+xml.xml                                                        x-appleworks-document.xml                                        x-killustrator.xml              x-sms-rom.xml
> relax-ng-compact-syntax.xml                                        x-applix-spreadsheet.xml                                         x-kivio.xml                     x-soap-remoting.xml
> relaxng.xml                                                        x-applix-word.xml                                                x-kmplot.xml                    x-source-rpm.xml
> rss+xml.xml                                                        x-archive.xml                                                    x-kns.xml                       xspf+xml.xml
> rtf.xml                                                            x-arc.xml                                                        x-kolf.xml                      x-spss-por.xml
> sdp.xml                                                            x-arj.xml                                                        x-kommander.xml                 x-spss-sav.xml
> sieve.xml                                                          x-asax.xml                                                       x-kontour.xml                   x-sqlite2.xml
> smil+xml.xml                                                       x-ascx.xml                                                       x-kopete-emoticons.xml          x-stuffit.xml
> sql.xml                                                            x-ashx.xml                                                       x-kourse.xml                    x-subrip.xml
> trig.xml                                                           x-asix.xml                                                       x-kpovmodeler.xml               x-superkaramba.xml
> vnd.adobe.flash.movie.xml                                          x-asmx.xml                                                       x-kpresenter.xml                x-sv4cpio.xml
> vnd.amazon.mobi8-ebook.xml                                         x-asp.xml                                                        x-krita.xml                     x-sv4crc.xml
> vnd.android.package-archive.xml                                    x-aspx.xml                                                       x-kseg.xml                      x-sysprof-capture.xml
> vnd.appimage.xml                                                   x-atari-2600-rom.xml                                             x-kspread-crypt.xml             x-t602.xml
> vnd.apple.keynote.xml                                              x-atari-7800-rom.xml                                             x-kspread.xml                   x-taisei-replay.xml
> vnd.apple.mpegurl.xml                                              x-atari-lynx-rom.xml                                             x-ksysguard.xml                 x-tar.xml
> vnd.chess-pgn.xml                                                  x-audacity-project.xml                                           x-ksysv-package.xml             x-tarz.xml
> vnd.coffeescript.xml                                               x-awk.xml                                                        x-ktheme.xml                    x-tex-gf.xml
> vnd.comicbook-rar.xml                                              x-axd.xml                                                        x-kudesigner.xml                x-tex-pk.xml
> vnd.comicbook+zip.xml                                              x-bcpio.xml                                                      x-kugar.xml                     x-tgif.xml
> vnd.corel-draw.xml                                                 x-biab.xml                                                       x-kvtml.xml                     x-theme.xml
> vnd.debian.binary-package.xml                                      x-bittorrent.xml                                                 x-kwallet.xml                   x-thomson-cartridge-memo7.xml
> vnd.emusic-emusic_package.xml                                      x-blender.xml                                                    x-kword-crypt.xml               x-thomson-cassette.xml
> vnd.flatpak.ref.xml                                                x-bps-patch.xml                                                  x-kwordquiz.xml                 x-thomson-sap-image.xml
> vnd.flatpak.repo.xml                                               x-bsdiff.xml                                                     x-kword.xml                     x-toutdoux.xml
> vnd.flatpak.xml                                                    x-bww.xml                                                        x-ldb.xml                       x-trash.xml
> vnd.font-fontforge-sfd.xml                                         x-bzdvi.xml                                                      x-lha.xml                       x-troff-man-compressed.xml
> vnd.framemaker.xml                                                 x-bzip-compressed-tar.xml                                        x-lhz.xml                       x-troff-man.xml
> vnd.google-earth.kml+xml.xml                                       x-bzip.xml                                                       xliff+xml.xml                   x-tuberling.xml
> vnd.google-earth.kmz.xml                                           x-bzpdf.xml                                                      x-lmt.xml                       x-turtle.xml
> vnd.hp-hpgl.xml                                                    x-bzpostscript.xml                                               x-lmu.xml                       x-tzo.xml
> vnd.hp-pcl.xml                                                     x-cabri.xml                                                      x-lrzip-compressed-tar.xml      x-ufraw.xml
> vnd.iccprofile.xml                                                 x-capella.xml                                                    x-lrzip.xml                     x-ugene-abi.xml
> vnd.kde.bluedevil-sendfile.xml                                     x-cb7.xml                                                        x-lsd.xml                       x-ugene-aln.xml
> vnd.kde.fontspackage.xml                                           x-cbt.xml                                                        x-lyx.xml                       x-ugene-db.xml
> vnd.kde.kcfgc.xml                                                  x-ccmx.xml                                                       x-lz4-compressed-tar.xml        x-ugene-embl.xml
> vnd.kde.kcfg.xml                                                   x-cda.xml                                                        x-lz4.xml                       x-ugene-fastq.xml
> vnd.kde.knotificationrc.xml                                        x-cd-image.xml                                                   x-lzip-compressed-tar.xml       x-ugene-fa.xml
> vnd.kde.kphotoalbum-import.xml                                     x-cdrdao-toc.xml                                                 x-lzip.xml                      x-ugene-gb.xml
> vnd.kde.kxmlguirc.xml                                              x-cisco-vpn-settings.xml                                         x-lzma-compressed-tar.xml       x-ugene-gff.xml
> vnd.kde.okular-archive.xml                                         x-class-file.xml                                                 x-lzma.xml                      x-ugene-hmm.xml
> vnd.lotus-1-2-3.xml                                                x-cmbx.xml                                                       x-lzop.xml                      x-ugene-mmdb.xml
> vnd.lotus-wordpro.xml                                              x-code-workspace.xml                                             x-lzpdf.xml                     x-ugene-msf.xml
> vnd.mlt+xml.xml                                                    x-compressed-iso.xml                                             x-m4.xml                        x-ugene-newick.xml
> vnd.mozilla.xul+xml.xml                                            x-compressed-tar.xml                                             x-macbinary.xml                 x-ugene-pdb.xml
> vnd.ms-access.xml                                                  x-compress.xml                                                   x-machine-config.xml            x-ugene-sam-bam.xml
> vnd.ms-asf.xml                                                     x-config.xml                                                     x-magicpoint.xml                x-ugene-scf.xml
> vnd.ms-cab-compressed.xml                                          x-core.xml                                                       x-mame-chd.xml                  x-ugene-srfa.xml
> vnd.ms-excel.addin.macroenabled.12.xml                             x-cpio-compressed.xml                                            x-markaby.xml                   x-ugene-sto.xml
> vnd.ms-excel.sheet.binary.macroenabled.12.xml                      x-cpio.xml                                                       x-master-page.xml               x-ugene-sw.xml
> vnd.ms-excel.sheet.macroenabled.12.xml                             x-csh.xml                                                        x-matroska.xml                  x-ugene-uprj.xml
> vnd.ms-excel.template.macroenabled.12.xml                          x-csproj.xml                                                     x-mdp.xml                       x-ugene-uql.xml
> vnd.ms-excel.xml                                                   x-cue.xml                                                        x-mds.xml                       x-ugene-uwl.xml
> vnd.ms-htmlhelp.xml                                                x-dar.xml                                                        x-mdsx.xml                      x-uml.xml
> vnd.ms-powerpoint.addin.macroenabled.12.xml                        x-dbf.xml                                                        x-mga-exec.xml                  x-urpmi-media.xml
> vnd.ms-powerpoint.presentation.macroenabled.12.xml                 x-designer.xml                                                   x-mif.xml                       x-urpmi.xml
> vnd.ms-powerpoint.slide.macroenabled.12.xml                        x-desktop.xml                                                    x-mimearchive.xml               x-ustar.xml
> vnd.ms-powerpoint.slideshow.macroenabled.12.xml                    x-dia-diagram.xml                                                xml-dtd.xml                     x-vbproj.xml
> vnd.ms-powerpoint.template.macroenabled.12.xml                     x-dia-shape.xml                                                  xml-external-parsed-entity.xml  x-virtualbox-hdd.xml
> vnd.ms-powerpoint.xml                                              x-discjuggler-cd-image.xml                                       xml.xml                         x-virtualbox-ova.xml
> vnd.ms-publisher.xml                                               x-disco.xml                                                      x-mobipocket-ebook.xml          x-virtualbox-ovf.xml
> vnd.ms-tnef.xml                                                    x-docbook+xml.xml                                                x-mozilla-bookmarks.xml         x-virtualbox-vbox-extpack.xml
> vnd.ms-visio.drawing.macroenabled.main+xml.xml                     x-doom-wad.xml                                                   x-ms-dos-executable.xml         x-virtualbox-vbox.xml
> vnd.ms-visio.drawing.main+xml.xml                                  x-dreamcast-rom.xml                                              x-msi.xml                       x-virtualbox-vdi.xml
> vnd.ms-visio.stencil.macroenabled.main+xml.xml                     x-drgeo.xml                                                      x-ms-shortcut.xml               x-virtualbox-vhd.xml
> vnd.ms-visio.stencil.main+xml.xml                                  x-dvi.xml                                                        x-ms-wim.xml                    x-virtualbox-vmdk.xml
> vnd.ms-visio.template.macroenabled.main+xml.xml                    x-egon.xml                                                       x-mswinurl.xml                  x-virtual-boy-rom.xml
> vnd.ms-visio.template.main+xml.xml                                 x-esri-shape-index.xml                                           x-mswrite.xml                   x-vnd.kde.kplato.work.xml
> vnd.ms-word.document.macroenabled.12.xml                           x-esri-shape.xml                                                 x-msx-rom.xml                   x-vnd.kde.kplato.xml
> vnd.ms-word.template.macroenabled.12.xml                           x-e-theme.xml                                                    x-musedata.xml                  x-vnd.kde.kugar.mixed.xml
> vnd.ms-word.xml                                                    x-executable.xml                                                 x-musescore.drumset+xml.xml     x-vnd.kde.plan.work.xml
> vnd.ms-works.xml                                                   x-fds-disk.xml                                                   x-musescore.palette.xml         x-vnd.kde.plan.xml
> vnd.ms-wpl.xml                                                     x-fictionbook+xml.xml                                            x-musescore.style+xml.xml       x-wais-source.xml
> vnd.ms-xpsdocument.xml                                             x-fluid.xml                                                      x-musescore.xml                 x-webarchive.xml
> vnd.nintendo.snes.rom.xml                                          x-font-afm.xml                                                   x-musescore+xml.xml             x-web-config.xml
> vnd.oasis.opendocument.chart-template.xml                          x-font-bdf.xml                                                   x-n64-rom.xml                   x-wii-rom.xml
> vnd.oasis.opendocument.chart.xml                                   x-font-dos.xml                                                   x-nautilus-link.xml             x-wii-wad.xml
> vnd.oasis.opendocument.database.xml                                x-font-framemaker.xml                                            x-navi-animation.xml            x-windows-themepack.xml
> vnd.oasis.opendocument.formula-template.xml                        x-font-libgrx.xml                                                x-neo-geo-pocket-color-rom.xml  x-wonderswan-color-rom.xml
> vnd.oasis.opendocument.formula.xml                                 x-font-linux-psf.xml                                             x-neo-geo-pocket-rom.xml        x-wonderswan-rom.xml
> vnd.oasis.opendocument.graphics-flat-xml.xml                       x-font-pcf.xml                                                   x-nes-rom.xml                   x-wpg.xml
> vnd.oasis.opendocument.graphics-template.xml                       x-font-snf.xml                                                   x-netcdf.xml                    x-wsdl.xml
> vnd.oasis.opendocument.graphics.xml                                x-font-speedo.xml                                                x-netshow-channel.xml           x-wwf.xml
> vnd.oasis.opendocument.image.xml                                   x-font-sunos-news.xml                                            x-nintendo-3ds-executable.xml   x-x509-ca-cert.xml
> vnd.oasis.opendocument.presentation-flat-xml.xml                   x-font-tex-tfm.xml                                               x-nintendo-3ds-rom.xml          x-xar.xml
> vnd.oasis.opendocument.presentation-template.xml                   x-font-tex.xml                                                   x-nintendo-ds-rom.xml           x-xbel.xml
> vnd.oasis.opendocument.presentation.xml                            x-font-ttx.xml                                                   x-nzb.xml                       x-xoj.xml
> vnd.oasis.opendocument.spreadsheet-flat-xml.xml                    x-font-type1.xml                                                 x-object.xml                    x-xpinstall.xml
> vnd.oasis.opendocument.spreadsheet-template.xml                    x-font-vfont.xml                                                 x-oleo.xml                      x-xz-compressed-tar.xml
> vnd.oasis.opendocument.spreadsheet.xml                             x-gameboy-color-rom.xml                                          x-ole-storage.xml               x-xzpdf.xml
> vnd.oasis.opendocument.text-flat-xml.xml                           x-gameboy-rom.xml                                                x-osm+xml.xml                   x-xz.xml
> vnd.oasis.opendocument.text-master-template.xml                    x-gamecube-rom.xml                                               x-overture.xml                  x-yaml.xml
> vnd.oasis.opendocument.text-master.xml                             x-gamegear-rom.xml                                               x-pagemaker.xml                 x-zerosize.xml
> vnd.oasis.opendocument.text-template.xml                           x-gba-rom.xml                                                    x-pak.xml                       x-zip-compressed-fb2.xml
> vnd.oasis.opendocument.text-web.xml                                x-gdbm.xml                                                       x-par2.xml                      x-zoo.xml
> vnd.oasis.opendocument.text.xml                                    x-gd-rom-cue.xml                                                 x-partial-download.xml          x-zstd-compressed-tar.xml
> vnd.openofficeorg.extension.xml                                    x-gedcom.xml                                                     x-pc-engine-rom.xml             zip.xml
> vnd.openshot-qt-project.xml                                        x-genesis-32x-rom.xml                                            x-pef-executable.xml            zlib.xml
> vnd.openxmlformats-officedocument.presentationml.presentation.xml  x-genesis-rom.xml                                                x-perl.xml                      zstd.xml
> vnd.openxmlformats-officedocument.presentationml.slideshow.xml     x-gettext-translation.xml                                        x-php.xml
> ```

You have to look in `/usr/share/packages`. `usr/share/applications` contains the single definition for every type extracted by `update-mime-database. Unlike the icons, I can't seem to find an official name convention, but in my `packages` directory there are two name conventions that seem to be followed:
 - The application name (eg. `lmms.xml`)
 - The application package name (eg. `org.openshot.OpenShot`)

Although they are a minority IMO it makes more sense to name it in the second way, but I guess this is more a matter of taste than anything else.

Reply:
That wasn't the case before either. For a quick fix we could write a little `README` or document it in the docs, but the most ideal fix would be to add an `install` function that does this automatically.